lib/plugins/helper/open_graph.js

```diff
@@ -59,12 +57,18 @@ function openGraphHelper(options = {}) {
   if (!images.length && content) {
     images = images.slice();
 
-    const $ = cheerio.load(content);
+    if (content.includes('<img')) {
+      const imgPattern = /<img.*?>/gi;
+      const srcPattern = /src=['"]?([^'"]*)['"]?/i;
+      const imgArray = content.match(imgPattern);
+      for (let img of imgArray) {
+        const src = img.match(srcPattern);
+        if (src[1]) {
```

Comment:
Since the img element requires the src attribute, the null check of String # match () will not be necessary in the normal system.
If there is an img element without src, it should be treated as abnormal, but there is room for consideration in error handling.

Reply:
Since it is using regex rather than a dom parser to match the `<img>` tag, so an `<img>` tag without src could happened. If there is an `<img>` tag matches `imgPattern` but without `src` attribute, `img.match(srcPattern)` will return `null`. So should I use `if (src)` rather than `if (src[1])`?

Reply:
If you want to ignore an `img` element with an empty `src`, the following changes will eliminate unnecessary checks.
`const srcPattern = /src=['"]?([^'"]+)['"]?/i;`
`if (src) images.push(src[1]); `

Reply:
OK, I will change to that.

Reply:
Thank you for the correction.
However, with just this one line, even if the value of `src` is empty, it will be pushed to `images`.
If the value of `src` is intended to be ignored if it is empty, you need to change the definition of the `srcPattern` variable.

Reply:
I have pushed a new commit with a new regex pattern:

```
/<img [^>]*src=['"]([^'"]+)([^>]*>)/gi
```

This one will only match the `<img>` tag with non-empty `src` attribute.